### PR TITLE
Rearrange docs

### LINKED ISSUE #672: Rearrange docs index page to clarify the extensions structure
Currently, we have docs only for the client extension. We should rewrite the [index.adoc](https://github.com/quarkiverse/quarkus-openapi-generator/blob/main/docs/modules/ROOT/pages/index.adoc) page to clarify that this is a multipurpose extension for OpenAPI and add parent sections for Client, Server, and Wiremock in the future.

### PR DESCRIPTION
**Pull request by mcruzdev**
Many thanks for submitting your Pull Request :heart:!

Fixes #672 

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](CONTRIBUTING.md)
- [ ] Your code is properly formatted according to [our code style](CONTRIBUTING.md#code-style)
- [ ] Pull Request title contains the target branch if not targeting main: `[0.9.x] Subject`
- [ ] Pull Request contains link to the issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-quarkus2` to backport the original PR to the `quarkus2` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Review by carlesarnal — APPROVED** [review]

**Review by ricardozanini — APPROVED** [review]
Amazing work!

File: docs/modules/ROOT/pages/client.adoc [hunk 1]
```diff
@@ -0,0 +1,168 @@
+= Quarkus - Openapi Generator - Client
+:extension-status: preview
+
+This extension is for REST code generation for client side only. If you're looking for code generation for the server side, please take a look at the xref:server.adoc[Quarkus Openapi Generator Server Extension].
```
Comment (hbelmiro) [inline docs/modules/ROOT/pages/client.adoc:4]:
```suggestion
This extension is for REST code generation for client side only. If you're looking for code generation for the server side, please take a look at the xref:server.adoc[Quarkus OpenAPI Generator Server Extension].
```

File: docs/modules/ROOT/pages/client.adoc [hunk 2]
```diff
@@ -0,0 +1,168 @@
+= Quarkus - Openapi Generator - Client
```
Comment (hbelmiro) [inline docs/modules/ROOT/pages/client.adoc:1]:
```suggestion
= Quarkus - OpenAPI Generator - Client
```

File: docs/modules/ROOT/pages/index.adoc [hunk 3]
```diff
@@ -1,178 +1,17 @@
-= Quarkus - Openapi Generator - Client
+= Quarkus - Openapi Generator
```
Comment (hbelmiro) [inline docs/modules/ROOT/pages/index.adoc:1]:
```suggestion
= Quarkus - OpenAPI Generator
```

File: docs/modules/ROOT/pages/index.adoc [hunk 4]
```diff
@@ -1,178 +1,17 @@
...
 :extension-status: preview
 
+This a multipurpose Quarkus' extension for generation of https://quarkus.io/guides/rest-client[Rest Clients] and xref:server.adoc[Server stubs] based on OpenAPI specification files.
 
-WARNING: This is the instructions for the latest SNAPSHOT version (main branch). Please, see the https://github.com/quarkiverse/quarkus-openapi-generator/blob/1.0.1/README.md[latest **released** documentation] if you are looking for instructions.
+Currently, we support two types of code generation:
 
-WARNING: Check versions 1.x.x if you're still using Quarkus 2. But be aware that we no longer support Quarkus 2. That means there are no updates planned for those versions.
+Quarkus - Openapi Generator - Client: xref:client.adoc[Refer to our documentation] on how to generate these clients.
```
Comment (hbelmiro) [inline docs/modules/ROOT/pages/index.adoc:8]:
```suggestion
Quarkus - OpenAPI Generator - Client: xref:client.adoc[Refer to our documentation] on how to generate these clients.
```

File: docs/modules/ROOT/pages/index.adoc [hunk 5]
```diff
@@ -1,178 +1,17 @@
...
+
+Quarkus - Openapi Generator - Server: xref:server.adoc[Follow our guidelines] to generate server stubs using Apicurio.
```
Comment (hbelmiro) [inline docs/modules/ROOT/pages/index.adoc:10]:
```suggestion
Quarkus - OpenAPI Generator - Server: xref:server.adoc[Follow our guidelines] to generate server stubs using Apicurio.
```

File: docs/modules/ROOT/pages/server.adoc [hunk 6]
```diff
@@ -0,0 +1,15 @@
+= Quarkus - Openapi Generator - Server
```
Comment (hbelmiro) [inline docs/modules/ROOT/pages/server.adoc:1]:
```suggestion
= Quarkus - OpenAPI Generator - Server
```

**Review by hbelmiro — COMMENTED** [review]
Great work @mcruzdev!
I just left a few suggestions to fix the OpenAPI case, that probably already existed before this PR.

**Comment by mcruzdev** [comment]
> Great work @mcruzdev! I just left a few suggestions to fix the OpenAPI case, that probably already existed before this PR.

Thank you @hbelmiro, I will squash soon!

**Review by hbelmiro — APPROVED** [review]
@mcruzdev We squash when we merge.
Can I merge it?

**Comment by mcruzdev** [comment]
> @mcruzdev We squash when we merge. Can I merge it?

Yes! Thank you for reviewing it!

**Pull request merged**